PixelGamEngineSolution/FlappyBird/Main.cpp

```diff
@@ -1,76 +1,273 @@
 
+	void DrawBird()
+	{
+		if (RedBird.GetVelocityY() > 0)
+		{
+			DrawDecal(RedBird.GetPosition(), UpDecal, { 0.12f, 0.12f });
+		}
+		else
+		{
+			DrawDecal(RedBird.GetPosition(), DownDecal, { 0.12f, 0.12f });
+		}
+	}
+	
+	// --------------------------------------------------------------------------------
 
-		if (GetKey(olc::SPACE).bPressed)
+	void BirdMechanics(const float InElapsedTime)
+	{
+		if (GetKey(olc::SPACE).bPressed && RedBird.GetVelocityY() >= Gravity / 6.0f)
 		{
-			BirdAcceleration_y = 0.0f;
-			BirdVelocity_y = -Gravity / 4.0f;
+			RedBird.Jump(Gravity);
+
+			FlapsCount++;
+			if (FlapsCount > MaxFlapCount)
+			{
+				MaxFlapCount = FlapsCount;
+			}
 		}
 		else
 		{
-			BirdAcceleration_y += Gravity * ElapsedTime;
+			RedBird.Fall(Gravity * InElapsedTime);
 		}
 
-		if (BirdAcceleration_y >= Gravity)
+		if (RedBird.GetAccelerationY() >= Gravity)
 		{
-			BirdAcceleration_y = Gravity;
+			RedBird.SetAccelerationY(Gravity);
 		}
 
-		//physics
-		BirdVelocity_y += BirdAcceleration_y * ElapsedTime;
-		BirdPosition_y += BirdVelocity_y * ElapsedTime;
+		RedBird.UpdatePosition(InElapsedTime);
 
-		int Bird_X = (int)(ScreenWidth() / 3.0f);//Sets the x position of the bird, this never changes
+		LevelPosition += 14.0f * InElapsedTime;
+	}
+	
+	// --------------------------------------------------------------------------------
 
-		//Draw bird
-		if (BirdVelocity_y > 0)
+	void SectionThing()
+	{
+		if (LevelPosition > SectionWidth + 10)
+		{
+			LevelPosition -= SectionWidth;
+			ListSection.pop_front();
+			int i = rand() % (ScreenHeight() - 20);
+
+			if (i <= 10)
+			{
+				i = 0;
+			}
+			ListSection.push_back(i);
+		}
+
+		int SectionIndex = 0;
+		for (int PipeSize : ListSection)
 		{
-			Draw(Bird_X, BirdPosition_y + 0, olc::WHITE);
-			Draw(Bird_X, BirdPosition_y + 1, olc::BLUE);
+			if (PipeSize != 0)
+			{
+				olc::vf2d PipePositionUp = { (float)(SectionIndex * SectionWidth + 20 - LevelPosition), (float)(ScreenHeight() - PipeSize - 20) };
+				olc::vf2d PipePositionDown = { (float)(SectionIndex * SectionWidth + 20 - LevelPosition), (float)(ScreenHeight() - PipeSize) };
+
+				DrawDecal(PipePositionUp, PipeDecal, { 0.15f, -0.15f });
+				DrawDecal(PipePositionDown, PipeDecal, { 0.15f, 0.15f });
+
+				olc::vf2d BirdPosition = RedBird.GetPosition();
+				//pipes collitions
+				if (BirdPosition.x > PipePositionUp.x && BirdPosition.x < PipePositionUp.x + 8 && BirdPosition.y < PipePositionUp.y)
+				{//upper pipe
+					bHasCollided = true;
+				}
+				if (BirdPosition.x > PipePositionDown.x && BirdPosition.x < PipePositionDown.x + 8 && BirdPosition.y > PipePositionDown.y)
+				{//Down pipe
+					bHasCollided = true;
+				}
+			}
+			SectionIndex++;
+		}
+	}
+	
+	// --------------------------------------------------------------------------------
+
+	void PrintNumber(const int InValue)
+	{
+		olc::vf2d Position = { ((float)ScreenWidth() / 2) - 2, 5 };
```

Comment:
Hmmm since this is always going to be the same position I'm assuming you only use this to print the flapcount.
So, a more accurate name for the method could be PrintFlapCount